### DISCUSSION TITLE
How do I compile neovim for windows on arm?

### DISCUSSION THREAD

**Discussion by asheryoung10** [neovim/neovim#29872] [Q&A]
I am on a new windows on arm laptop and would like to use neovim. ~~I have tried the x86 version on the releases page but am running into treesitter issues that I think are caused by the architectural difference.~~ (if not I still want to get a native binary) I am fairly inept when it comes to cmake and much of what I have tried so far has been AI instructions.

I have tried to compile neovim for windows on arm with  the lllvm-mingw-20240619-ucrt-aarch64 toolchain ([aarch64-w64-mingw32](https://github.com/mstorsjo/llvm-mingw) clang) and with visual studio:

With the mingw clang toolchain, cmake fails to automatically retrieve dependencies like luv, I tried fixing that for a while and gave up. 
With visual studio, the same thing happens if I run cmake in the visual studio command prompt. 

The only time cmake has actually retrieved dependencies has been after I cloned the project with visual studio and configured cmake with visual studio's built in tooling, however with this method I can't find out how to set the architecture to arm, visual studio only enumerates x86_64 build options and for a reason beyond me overwrites any manual attempts at setting the target to arm with x64 builds. 

How should I go about building for windows on arm?
[Closed similar discussion from four years ago](https://github.com/neovim/neovim/issues/12521)

~~the mentioned tree sitter error on amd64 binary running on arm (might be unrelated): (Could not load parser for c: "...win64/share/nvim/runtime/lua/vim/treesitter/language.lua:112: Failed to load parser for language 'c': uv_dlopen: no error" )~~(fixed that error, treesitter was compiling parsers with my native arm compiler and couldn't load them into my x86_64 neovim process)

**Comment by Daydreamer-riri** [comment]
There are some related discussions here: https://github.com/neovim/neovim/issues/29416. I'm also curious to know how to use NVIM on my ARM computer.
